plugin/storage/es/dependencystore/storage.go

```diff
@@ -21,27 +21,123 @@
+const (
+	dependencyType        = "dependencies"
+	dependencyIndexPrefix = "jaeger-dependencies-"
+)
+
+type timeToDependencies struct {
+	Timestamp    time.Time              `json:"timestamp"`
+	Dependencies []model.DependencyLink `json:"dependencies"`
+}
 
 // DependencyStore handles all queries and insertions to ElasticSearch dependencies
-type DependencyStore struct{}
+type DependencyStore struct {
+	ctx    context.Context
+	client es.Client
+	logger *zap.Logger
+}
 
 // NewDependencyStore returns a DependencyStore
-func NewDependencyStore() *DependencyStore {
-	return &DependencyStore{}
+func NewDependencyStore(client es.Client, logger *zap.Logger) *DependencyStore {
+	return &DependencyStore{
+		ctx:    context.Background(),
+		client: client,
+		logger: logger,
+	}
 }
 
 // WriteDependencies implements dependencystore.Writer#WriteDependencies.
 func (s *DependencyStore) WriteDependencies(ts time.Time, dependencies []model.DependencyLink) error {
+	indexName := indexName(ts)
+	if err := s.createIndex(indexName); err != nil {
+		return err
+	}
+	if err := s.writeDependencies(indexName, ts, dependencies); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (s *DependencyStore) createIndex(indexName string) error {
+	_, err := s.client.CreateIndex(indexName).Body(dependenciesMapping).Do(s.ctx)
+	if err != nil {
+		return errors.Wrap(err, "Failed to create index")
+	}
+	return nil
+}
+
+func (s *DependencyStore) writeDependencies(indexName string, ts time.Time, dependencies []model.DependencyLink) error {
+	_, err := s.client.Index().Index(indexName).
+		Type(dependencyType).
+		BodyJson(&timeToDependencies{
+			Timestamp:    ts,
+			Dependencies: dependencies,
+		}).
+		Do(s.ctx)
+	if err != nil {
+		return errors.Wrap(err, "Failed to write dependencies")
+	}
 	return nil
 }
 
 // GetDependencies returns all interservice dependencies
 func (s *DependencyStore) GetDependencies(endTs time.Time, lookback time.Duration) ([]model.DependencyLink, error) {
-	return nil, nil
+	searchResult, err := s.client.Search(getIndices(endTs, lookback)...).
+		Type(dependencyType).
+		Size(10000). // the default elasticsearch allowed limit
+		Query(buildTSQuery(endTs, lookback)).
+		Do(s.ctx)
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed to search for dependencies")
+	}
+
+	var retDependencies []model.DependencyLink
+	hits := searchResult.Hits.Hits
+	for _, hit := range hits {
+		source := hit.Source
+		var tToD timeToDependencies
+		if err := json.Unmarshal(*source, &tToD); err != nil {
+			return nil, errors.New("Unmarshalling ElasticSearch documents failed")
+		}
+		retDependencies = append(retDependencies, tToD.Dependencies...)
+	}
+	return retDependencies, nil
+}
+
+func buildTSQuery(endTs time.Time, lookback time.Duration) elastic.Query {
+	return elastic.NewRangeQuery("timestamp").Gte(endTs.Add(-lookback)).Lte(endTs)
+}
+
+func getIndices(ts time.Time, lookback time.Duration) []string {
+	var indices []string
```

Comment:
why doesn't the previous code you had here work? What edge case did it not cover that this does?